server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -161,39 +172,25 @@ public void getSynonymRules(String resourceName, int from, int size, ActionListe
             .setQuery(QueryBuilders.termQuery(SYNONYMS_SET_FIELD, resourceName))
             .setFrom(from)
             .setSize(size)
+            .addSort("id", SortOrder.ASC)
             .setPreference(Preference.LOCAL.type())
             .setTrackTotalHits(true)
-            .execute(new ActionListener<>() {
-                @Override
-                public void onResponse(SearchResponse searchResponse) {
-                    final long totalSynonymRules = searchResponse.getHits().getTotalHits().value;
-                    if (totalSynonymRules == 0) {
-                        listener.onFailure(new ResourceNotFoundException("Synonym set [" + resourceName + "] not found"));
-                        return;
-                    }
-                    final SynonymRule[] synonymRules = Arrays.stream(searchResponse.getHits().getHits())
-                        .map(SynonymsManagementAPIService::hitToSynonymRule)
-                        .toArray(SynonymRule[]::new);
-                    listener.onResponse(new PagedResult<>(totalSynonymRules, synonymRules));
-                }
-
-                @Override
-                public void onFailure(Exception e) {
-                    final Throwable cause = ExceptionsHelper.unwrapCause(e);
-                    if (cause instanceof IndexNotFoundException) {
-                        listener.onFailure(new ResourceNotFoundException("Synonym set [" + resourceName + "] not found"));
-                        return;
-                    }
-                    listener.onFailure(e);
+            .execute(new DelegatingIndexNotFoundActionListener<>(resourceName, listener, (l, searchResponse) -> {
```

Comment:
Used a  `DelegatingIndexNotFoundActionListener` class to ensure that when index is not found, it is translated into a `ResourceNotFound`. I'd like to have reused the class from Enterprise Search, but given it's in XPack, I didn't want to push things here.

If anyone can think of a good place for this class to be, I'm up for moving it to a top-level class!

Reply:
I am not sure if it is necessary to introduce this class for this small single change, but I will leave this to you.